tests/conftest.py

```diff
@@ -28,48 +34,41 @@ def default_bbr() -> RuleCollection:
     return RuleCollection.default_bbr()
 
 
-class BaseRuleTest(unittest.TestCase):
+@pytest.fixture(scope="class")
```

Comment:
Noticing how this fixture is in `conftest.py` and loads all the rules so is pretty expensive, should we expand the fixture scope to `module` or `session` as it will be used by most other tests across testing?

Reply:
I think that makes sense, especially for the `module` level (in case customers would want to add their own test module which may have a different default configuration in the future). 

Trying to thing of a circumstance where we would want to specify different filters or other constants and thus would want to use class. Given that the `production_filter` right now is in effect a constant, I do not see an issue with expanding it. However, if the test classes were to specify their own filters (which I do not think we do right now), then we would probably want to keep it at `class`

Reply:
Good point Eric. I was thinking we may even be able to make a fixture for each "ruleset" with `autouse` set and the scope being `session`. Thus making them accessible anywhere in the testing but not one giant fixture. May be useful later down the road when we start applying marks?

The only confusion I have is...do we need to be so reliant on inheritance of `TestBaseRule` in the rest of the tests? If I understand correctly, we load all rules via a single fixture, use that in `TestBaseRule` and then rely on inheritance in every other testing file? If we are moving forward with pytest and fixtures arranging data, then we should be able to avoid this all together?

Reply:
I agree, I think the original purpose for the inheritance was due to the class extending `unittest.TestCase` to support the `unittest` framework and having that be inherited. With Pytest I do not think that is necessary.

Reply:
This PR is just a port. The changes are big enough as it is. I've updated the META to better reflect how we can chunk up this refactor into smaller PRs. One of which we'll iterate over different ways to "setup" the classes.

I started adding TODOs for other refactors we may want to consider, but I think this should all come after this PR.